### PR TITLE
feat: pep 621 page, flit

### PR DESCRIPTION
**Pull request by henryiii**
In honor of Flit joining the PyPA, here's a new version of our setup for pure Python projects.

### CODE REVIEW

File: pages/developers/packaging.md [hunk 1]
```diff
@@ -260,10 +260,7 @@ classifiers =
     Operating System :: Unix
     Programming Language :: C++
     Programming Language :: Python
-    Programming Language :: Python :: 2
-    Programming Language :: Python :: 2.7
```
Comment (eduardo-rodrigues) [inline pages/developers/packaging.md:264]: If you remove all these lines you might as well use Python 3 *only*

File: pages/developers/pep621.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+layout: page
+title: Pure Python packaging
+permalink: /developer/pep621
+nav_order: 4
+parent: Developer information
+---
+
+# Pure Python packaging
+
+Pure Python packages with simple builds, that is, no binary extensions, then
```
Comment (eduardo-rodrigues) [inline pages/developers/pep621.md:11]: This first sentene does not read well at all. It's super chatty.

File: pages/developers/pep621.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  "pytest >=6.0",
+]
+mpl = [
+  "matplotlib >=2.0",
+]
+```
+
+Self dependencies can be used by using the name of the package, such as `dev =
+["package[test,examples]"]`, but this requires Pip 21.2 or newer. We recommend
+providing at least `test`, `docs`, and `dev`.
```
Comment (eduardo-rodrigues) [inline pages/developers/pep621.md:139]: You mean test -> tests here, right?

File: pages/developers/pep621.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+inside it, such as `src/<package>/`.  This may seem like extra hassle; after
+all, you can type "`python`" in the main directory and avoid installing it if
+you don't have a `src` folder! However, this is a bad practice, and it causes
+several common bugs, such as running `pytest` and getting the local version
+instead of the installed version - this obviously tends to break if you build
+parts of the library or if you access package metadata.
+
+This sadly is not part of the standard metadata in `[project]`, so it depends
+on what backend you you use. Flit and PDM use automatic detection, while
+trampolim and whey do not, requiring a `tool` setting.
```
Comment (eduardo-rodrigues) [inline pages/developers/pep621.md:101]:
```suggestion
Trampolim and whey do not, requiring a `tool` setting.
```

File: pages/developers/pep621.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+
+## Versioning
+
+You can specify the version manually (as shown in the example), but the
+backends usually provide some automatic features to help you avoid this. Flit
+will pull this from a file if you ask it to. PDM can be instructed to look in a
+file or use git. Trampolim can be instructed to use git.
```
Comment (eduardo-rodrigues) [inline pages/developers/pep621.md:109]:
```suggestion
file or use git. Trampolim can be instructed to use Git.
```

File: pages/developers/pep621.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+[project.urls]
+Source = "https://github.com/scikit-hep/package"
+Documentation = "https://package.readthedocs.io/"
+Bug Tracker = "https://github.com/scikit-hep/package/issues"
+Discussions = "https://github.com/scikit-hep/package/discussions"
+Changelog = "https://package.readthedocs.io/en/latest/changelog.html"
+```
+
+You can read more about each field, and all allowed fields, in [PEP 621][],
+or [Flit](https://flit.readthedocs.io/en/latest/pyproject_toml.html#new-style-metadata), or
```
Comment (eduardo-rodrigues) [inline pages/developers/pep621.md:86]:
```suggestion
[Flit](https://flit.readthedocs.io/en/latest/pyproject_toml.html#new-style-metadata), or
```

**Review by eduardo-rodrigues — APPROVED** [review]

File: pages/developers/packaging.md (hunk 1 above)
Comment (henryiii) [reply pages/developers/packaging.md:264]: Good point!

File: pages/developers/pep621.md (hunk 3 above)
Comment (henryiii) [reply pages/developers/pep621.md:139]: We generally use `[test]` for the extra, and `/tests` for the folder.

https://github.com/scikit-hep/boost-histogram/blob/928d066618ef45ddb3c6e64d515823a9a815520f/setup.py#L51-L64

Comment (eduardo-rodrigues) [reply pages/developers/pep621.md:139]: Indeed. Sorry, got the scope wrong. Agreed.

File: pages/developers/pep621.md (hunk 2 above)
Comment (henryiii) [reply pages/developers/pep621.md]: Redid the paragraph. Not sure what happened to this first sentence.

**Pull request merged**